shared/src/commonMain/kotlin/com/mbta/tid/mbta_app/model/NearbyStaticData.kt

```diff
@@ -449,19 +450,22 @@ data class NearbyStaticData(val data: List<TransitWithStops>) {
 
 /**
  * Attaches [schedules] and [predictions] to the route, stop, and routePattern to which they apply.
- * Removes non-typical route patterns which are not predicted within 90 minutes of [filterAtTime].
- * Sorts routes by subway first then nearest stop, stops by distance, and headsigns by route pattern
- * sort order.
+ * Removes non-typical route patterns which are not happening either at all or between
+ * [filterAtTime] and [filterAtTime] + [hideNonTypicalPatternsBeyondNext]. Sorts routes by subway
+ * first then nearest stop, stops by distance, and headsigns by route pattern sort order.
  *
  * Runs static data and predictions through [TemporaryTerminalRewriter].
  */
 fun NearbyStaticData.withRealtimeInfo(
     globalData: GlobalResponse?,
-    sortByDistanceFrom: Position,
+    sortByDistanceFrom: Position?,
     schedules: ScheduleResponse?,
     predictions: PredictionsStreamDataResponse?,
     alerts: AlertsStreamDataResponse?,
     filterAtTime: Instant,
+    showAllPatternsWhileLoading: Boolean,
+    hideNonTypicalPatternsBeyondNext: Duration?,
+    filterCancellations: Boolean,
```

Comment:
suggestion: It would be nice to have some unit tests for this new constructor / new parameters directly.

Reply:
Added a couple